### ISSUE TITLE
Size of empty array is not known statically?

### ISSUE THREAD

**Issue by Morriar** [sorbet/sorbet#2151]
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0Aclass%20A%0A%20%20extend%20T%3A%3ASig%0A%0A%20%20sig%20%7Bparams(x%3A%20Integer).void%7D%0A%20%20def%20self.foo(*x)%0A%20%20%20%20puts%20x.to_s%0A%20%20end%0Aend%0A%0Adef%20main%0A%20%20arr%20%3D%20%5B1%2C%202%2C%203%5D%0A%20%20A.foo(*arr)%20%23%20Ok%0A%0A%20%20arr%20%3D%20%5B%5D%0A%20%20A.foo(*arr)%20%23%20Seems%20like%20we%20can%20infer%20this%20size%20too%3F%20%20%0Aend)

```ruby
# typed: true
class A
  extend T::Sig

  sig {params(x: Integer).void}
  def self.foo(*x)
    puts x.to_s
  end
end

def main
  arr = [1, 2, 3]
  A.foo(*arr) # Ok

  arr = []
  A.foo(*arr) # Seems like we can infer this size too?  
end
```

#### Observed output

```
editor.rb:16: Splats are only supported where the size of the array is known statically https://srb.help/7019
    16 |  A.foo(*arr) # Seems like we can infer this size too?  
          ^^^^^^^^^^^^
Errors: 1
```

#### Expected behavior

While empty, the size is known statically so it should not error.

**Comment by axyjo** [comment]
The first `arr` isn't a `T::Array[Integer]`, it's inferred as a `[Integer, Integer, Integer]` tuple type. IIRC, Sorbet doesn't have array-length checks. [Demo](https://sorbet.run/#%23%20typed%3A%20true%0Aclass%20A%0A%20%20extend%20T%3A%3ASig%0A%0A%20%20sig%20%7Bparams(x%3A%20Integer).void%7D%0A%20%20def%20self.foo(*x)%0A%20%20%20%20puts%20x.to_s%0A%20%20end%0Aend%0A%0Adef%20main%0A%20%20arr%20%3D%20T.let(%5B1%2C%202%2C%203%5D%2C%20%5BInteger%2C%20Integer%2C%20Integer%5D)%0A%20%20A.foo(*arr)%20%23%20Ok%0A%0A%20%20arr%20%3D%20T.let(%5B1%2C%202%2C%203%5D%2C%20T%3A%3AArray%5BInteger%5D)%0A%20%20A.foo(*arr)%0A%0A%20%20arr%20%3D%20%5B%5D%0A%20%20A.foo(*arr)%20%23%20Seems%20like%20we%20can%20infer%20this%20size%20too%3F%20%20%0Aend)

**Comment by Morriar** [comment]
No more error 🎉

**Comment by jez** [comment]
(Fixed in #3836)
